### PR TITLE
Update the documentation to new API

### PR DESCRIPTION
**Pull request by frapac**

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -1,17 +1,17 @@
 # ExaPF
 
 [`ExaPF.jl`](https://github.com/exanauts/ExaPF.jl) is a
-package to solve the power flow problem on upcoming exascale architectures by solving a system of nonlinear equations and provide derivative information used for example in a reduced space optimization method.
```
Comment (michel2323) [inline docs/src/index.md:4]: is a differentiable power flow solver providing up to second-order derivatives? I know you mention it below. But for those who read only one sentence.

File: docs/src/lib/autodiff.md [hunk 2]
```diff
@@ -3,16 +3,30 @@ CurrentModule = ExaPF.AutoDiff
...
+
+
+## First and second-order derivatives
 
 ```@docs
+AbstractJacobian
 AbstractHessianProd
 AbstractFullHessian
-AbstractJacobian
-AbstractStack
```
Comment (michel2323) [inline docs/src/lib/autodiff.md:12]: No more stack?

File: docs/src/quickstart.md [hunk 3]
```diff
@@ -45,13 +45,14 @@ datafile = joinpath(artifact_path(exadata_hash), "ExaData", "case1354.m")
...
-convergence = ExaPF.powerflow(polar, pf_algo)
+polar = ExaPF.PolarForm(datafile, CPU())  # Load data
+stack = ExaPF.NetworkStack(polar)         # Load variables
+convergence = run_pf(polar, stack; verbose=1)
+
 ```
 
 Implicitly, ExaPF has just proceed to the following operations:
-- instantiate automatically a starting point ``x_0`` from MATPOWER's data
+- instantiate automatically a starting point ``x_0`` from the variables stores in `stack`
```
Comment (michel2323) [inline docs/src/quickstart.md:55]: stored

File: docs/src/quickstart.md [hunk 4]
```diff
@@ -74,14 +75,14 @@ of the network. For instance, we can retrieve the number of buses
...
 ```
 
 However, a [`ExaPF.PowerSystem.PowerNetwork`](@ref) object stores only the **physical** attributes
-of the network, independently of the mathematical formulations
-we can use to model the network. To choose a particular formulation,
+of the network.
+To choose a given mathematical formulation,
 we need to pass the object `pf` to an [`ExaPF.AbstractFormulation`](@ref) layer.
-Currently, the only layer implemented is the polar formulation,
+Currently, only the the polar formulation is provided
```
Comment (michel2323) [inline docs/src/quickstart.md:85]: the the

**Review by michel2323 — APPROVED** [review]

File: docs/src/lib/autodiff.md (hunk 2 above)
Comment (frapac) [reply docs/src/lib/autodiff.md:12]: The stack has been moved right before in the file `autodiff.md` :)

File: docs/src/index.md (hunk 1 above)
Comment (frapac) [reply docs/src/index.md:4]: indeed, good suggestion

**Pull request merged**